core/lib/dal/migrations/20231213192041_applied-snapshot-status.up.sql

```diff
@@ -0,0 +1,6 @@
+CREATE TABLE applied_snapshot_status
+(
+    l1_batch_number         BIGINT NOT NULL PRIMARY KEY,
+    is_finished             BOOL NOT NULL,
```

Comment:
In vacuum, I think that a status column modeling an enum would be better, since it'd allow for a richer lifecycle and allow extending it in the future.

Reply:
Enum is better than bool, but perhaps in this case we can get away without a status column at all - having both `last_finished_chunk_id` and `last_chunk_id` we can infer this info from the DB - in this case `status` is denormalization. 
Although it's less explicit so I won't insist.

Also, I think we should mark snapshot recovery as done as soon as the postgres recolvery stage is complete. At this point other components might can start - as we discussed, they must not rely on any l1 batches present in the system. Or do you guys already invision a more extensive lifecycle of rows in this table?

Reply:
Do you mean inferring the recovery status from `last_finished_chunk_id` and `total_number_of_chunks`? I think this is OK, at least at this point. There's no need to invent complexities where we can avoid them.

Reply:
> Also, I think we should mark snapshot recovery as done as soon as the postgres recolvery stage is complete (...)
I agree with that part.
I also agree that we could infer is_finished from last_finished_chunk_id and total_chunk_count, but I am a big fan of denormalization in such cases (i.e. a single column that contains "status" of the whole row)

Reply:
I still think it's better not to denormalize unless we have a good reason to, which I don't see here. 
But it's not a strong opinion - it's also more explicit with such a column, so feel free to do either way